src/script/sign.cpp

```diff
@@ -33,14 +33,60 @@ bool MutableTransactionSignatureCreator::CreateSig(const SigningProvider& provid
     return true;
 }
 
+static bool GetCScript(const SigningProvider* provider, const SignatureData* sigdata, const CScriptID &scriptid, CScript& script)
+{
+    if (provider != nullptr && provider->GetCScript(scriptid, script)) {
+        return true;
+    }
+    // Look for scripts in SignatureData
+    if (CScriptID(sigdata->redeem_script) == scriptid) {
+        script = sigdata->redeem_script;
+        return true;
+    } else if (CScriptID(sigdata->witness_script) == scriptid) {
+        script = sigdata->witness_script;
+        return true;
+    }
+    return false;
+}
+
+static bool GetPubKey(const SigningProvider* provider, const SignatureData* sigdata, const CKeyID &address, CPubKey& pubkey)
+{
+    if (provider != nullptr && provider->GetPubKey(address, pubkey)) {
+        return true;
+    }
+    // Look for pubkey in all partial sigs
+    const auto& it = sigdata->signatures.find(address);
```

Comment:
Commit "Replace CombineSignatures with ProduceSignature"

As pointed by @MarcoFalke, don't use references to iterators. Check throughout too.

Reply:
Done.